### DISCUSSION TITLE
How to make time delay within certain hours of each day?

### DISCUSSION THREAD

**Discussion by kelp-videos** [gopro/labs#798] [Q&A]
Hi team,
I have been using the QR code creator to program my gopro to take a time-lapse from 7am-7pm, with one photo every 5
 minutes. I also have the gopro connected to a lithium ion battery to sustain charge while keeping the original gopro battery in (at 100%). I did a test run and there were no photos recorded on the sd card even though the programming still said it would wake up at 7am. (I left it for three days and it did not record anything).

I would like to try setting the code to take one photo every 5 minutes (on repeat), but only from 7am-7pm, and repeat this command everyday. I'm not a coder and was wondering if someone knew how to code for this setting because it is not possible on the interactive QR code creator website.

![Screenshot (91)](https://github.com/gopro/labs/assets/157651264/62a94f78-d7fb-4abe-adf9-740f0c9b5fd6)
![Screenshot (92)](https://github.com/gopro/labs/assets/157651264/ed037956-3c21-409e-93f4-46159d816852)

**Comment by dnewman-gpsw** [comment]
Nothing wrong with that command if you followed this instruction (in your screen capture)

```Time/Night-Lapse modes: The camera defaults to having presets for only one time-lapse mode and one night-lapse mode, yet both modes have two configurations (video or photo.) For all QR Code mode commands to work, a preset is needed for every mode you wish to use. e.g. If you only have a preset for night-lapse video (NLV), the QR command for NLV will work, but will not for night-lapse photo (NLP). To solve this, create a new preset for NLP, and both video and photo night-lapse commands will work.```

I expect you don't have a Timelapse Photo Preset enabled on your camera.

**Reply by kelp-videos** [reply]
I set the default as a timelapse.

**Reply by kelp-videos** [reply]
Also I wonder if the gopro does not carry out the commands when it is charging???

**Reply by dnewman-gpsw** [reply]
Re-read, you have a missing preset.

**Reply by kelp-videos** [reply]
What is my missing preset?

**Reply by dnewman-gpsw** [reply]
"The camera defaults to having presets for only **one time-lapse** mode and one night-lapse mode, yet both modes have two configurations (video or photo.) "  

You need to create a Photo Time Lapse preset

**Reply by kelp-videos** [reply]
The presets do not allow me to select both the photo and the timelapse mode

**Reply by kelp-videos** [reply]
Also, do you think that plugging the gorpro into constant power source could make the pre-programming malfunction? (because it is charging) somehow?

**Reply by dnewman-gpsw** [reply]
> The presets do not allow me to select both the photo and the timelapse mode? 

Yes they do.  Edit the timelapse preset and select photo.

No, external power is not an issue.

**Reply by kelp-videos** [reply]
Can that only be done on the physical gopro or can you program it on the QR code?

**Reply by dnewman-gpsw** [reply]
It is one of the few things that can only be done through the camera menus.  Like the quoted text says, "create a new preset for NLP, and both video and photo night-lapse commands will work."  Same is true for timelapse mode.  Point is the QR code can select presets, but it can't select presets your camera doesn't have. If you create the missing preset it will work.

**Reply by kelp-videos** [reply]
Thank you for all your help. I was wondering one more thing. Is there a way to set the gopro to take footage from 7am-7pm, AND stop at a certain date (like pre program for March 10 for example, or program to stop after 10 days of recording)?

**Reply by dnewman-gpsw** [reply]
Yes, on HERO11 & 12 you can do more complex scripting. Use the visual code tool https://gopro.github.io/labs/build/ 
![image](https://github.com/gopro/labs/assets/10856084/ece0ae0b-b3f7-43bc-a0cf-fdbe5ab0b2c8)

Example script above would stop taking photos after March 10.

**Reply by kelp-videos** [reply]
This is so helpful, thank you! Also, is there a way to add "take photo every 10 minutes" within these time and date parameters?

**Reply by dnewman-gpsw** [reply]
Yes.

![image](https://github.com/gopro/labs/assets/10856084/087c54e8-e053-4d36-9b68-89c7bdc36d87)

![Time-Data Limited 10-minute Photos-3_11_2024, 13_21_21](https://github.com/gopro/labs/assets/10856084/380692d0-b1cb-42d8-b025-a389491b4fcb)

**Reply by kelp-videos** [reply]
Hi there, I just did a trial run of this QR code setup on my gopro and it was not reliably recording videos. I am now trying this setup, let me know if you see any problems in the block coding. Or if you have a full tutorial link on all the rules behind this blockly so I could understand it better, I would appreciate it!
![Screenshot (132)](https://github.com/gopro/labs/assets/157651264/8dfcef3d-ed08-47bf-8da7-009bb5de2c51)

**Reply by kelp-videos** [reply]
Also, I did the timelapse settings from 7am-7pm and it worked with the "sleep until 7am" block (first photo), but didn't seem to work without the "sleep until 7am" block (second photo). Why is this?
![Screenshot (139)](https://github.com/gopro/labs/assets/157651264/5a5b10d4-df3f-4e43-81d3-ecb54cc3054d)
![Screenshot (140)](https://github.com/gopro/labs/assets/157651264/3ee3bf9f-9a60-4498-bf24-17b0c520c2fd)

**Reply by dnewman-gpsw** [reply]
This is coding, so you need to define clearly what you want the camera to do.  Any coding friends?  I really don't know what you want it to do, and these images don't clarify it. You still have strange date limitations, but it will be best to start very simple and expand from there.

**Reply by kelp-videos** [reply]
Hi there, I have used the blockly code you provided and it did not record any photos, despite the screen confirming that it is scheduled to take the next photo in 10 minutes. Could you possibly make another blockly code that is scheduled to either take one photo every 20 minutes (or start timelapse photo mode) from 7am-7pm and turn off after April 4? I was wondering if the month and time aspects of the blockly code are throwing it off? I would appreciate an example of what should work for these parameters.

**Reply by dnewman-gpsw** [reply]
Why have it turn off after a particular date?  Why make it more complex?  There is are already examples the do 7am-7pm with timelapse https://gopro.github.io/labs/control/construction/

**Reply by kelp-videos** [reply]
We want to leave them underwater, attached to a Blue Robotics lithium ion external battery but do not want to drain the battery before we dive back down to retrieve them. Also, with the feature you showed, how do you know the interval the photos will be taken at?

**Reply by kelp-videos** [reply]
Also, I just tried the "daily long timelapse" and it is only flashing the message "Pause of 1 secs button to cancel" and says "labs processing" continuously...

**Reply by kelp-videos** [reply]
Also, I cannot find the blockly gorpo page anymore, has it been removed?

**Reply by dnewman-gpsw** [reply]
Why does it matter if the battery is drained?  It shouldn't be wired to kill the battery.  Once the battery has insufficient voltage to power the camera, the camera will not draw any more power, external battery will be fine.

> "daily long timelapse" and it is only flashing the message "Pause of 1 secs button to cancel" ...

As these script have worked for years, this is likely something you have set up wrong.  You need to share the codes, as I can't determine anything from this post.

Not moved still at https://gopro.github.io/labs/build/ and link from https://gopro.github.io/labs

**Comment by kelp-videos** [comment]
Thank you! I will test it out

**Comment by Saraiva09** [comment]
Hi,  I'm having more or less the same problem.

I want to set the camera to record multiple time lapse videos with a 5sec interval and 5min duration. I want the camera to record just between 07:00 and 21:00, with 15min gaps between each video, but I'm having some issues.

Unfortunately, I just have a gopro 9 so I can't use blockly. 

Is there a way to do it in go pro labs?

Thanks!

**Reply by dnewman-gpsw** [reply]
Yes, manually, although you can use Blockly to learn, there are only a few commands that the HERO9 doesn't support.  Try it.  Blockly was only to make it a little easier, but users for the first 3 years did this without Blockly, you can too.

**Reply by Saraiva09** [reply]
Thanks for the reply!

Already tried different codes but I can't figure it out.

The idea is to deploy the camera in a remote seamount to study a seasonal agregation of devil rays. So we are using an external power-source for the battery to last at least two weeks.

In the previus comment I forgot to mention that I need to repeat the cycle every day.

The last code I tried was this - !07:00S!900NmTp.5r1080fW0dR300!S!21:00E!R - but didn't work.

Do you think something like this will? - >=07:00<21:00mTp.5r1080fW0dR300!S!900N!R

**Reply by dnewman-gpsw** [reply]
Start one command at a time.

Try:  "Hello"!1R
Then:  >07:00"HELLO"!R
etc. 

Build up from there.

**Comment by caden-fluegge** [comment]
Hi! I'm trying to do something similar for long-term marine camera deployment.  This thread was very helpful.  kelp-videos, I saw that you mentioned using Blue Robotics lithium ion batteries above, and was curious about what housing/method of connection to the camera itself you are using.  Any help would be appreciated!

**Reply by kelp-videos** [reply]
Hi! This is an overview of the setup. We have gopro 11s, with sandisk V30 256GB micro sd cards, DC to DC converter (DC 12-24V input) and (5V or 3A output), and Blue Robotics 14.8V lithium-ion battery. There are a few extra attachments and soldering that are required, but the links are below:

**Reply by kelp-videos** [reply]
https://bluerobotics.com/store/comm-control-power/powersupplies-batteries/battery-li-4s-18ah-r3/

**Reply by kelp-videos** [reply]
https://www.amazon.ca/Converter-Charging-Waterproof-Compatible-Raspberry/dp/B0BNQ5JNWZ/ref=sr_1_17?crid=P3CRXFPJI0O6&keywords=12V%2B36V%2B48V%2Bto%2B5V%2BConverter%2BDC%2BModule%2BAdapter%2B10-55V%2Bto%2B5V%2BUSB-C&qid=1707932227&s=industrial&sprefix=12v%2B36v%2B48v%2Bto%2B5v%2Bconverter%2Bdc%2Bmodule%2Badapter%2B10-55v%2Bto%2B5v%2Busb-c%2B%2Cindustrial%2C121&sr=1-17&th=1

**Reply by kelp-videos** [reply]
https://www.amazon.ca/GELRHONR-Extension-Connector-Transmission-Phone-2Pack/dp/B09ZTNPT3B/ref=sr_1_6?crid=7VL3M2LX7SNV&keywords=right+corner+bent+usb+c+connecter&qid=1707932709&s=industrial&sprefix=right+corner+bent+usc+c+connecter%2Cindustrial%2C129&sr=1-6

**Reply by kelp-videos** [reply]
https://www.amazon.ca/XT90-S-Connector-Anti-Spark-Battery-Charger/dp/B074PTHZ3M/ref=sr_1_2_sspa?crid=23XZAEVXGP1OL&dib=eyJ2IjoiMSJ9.Ij_GmiEJlvhMCsWycGEqKT_he96ZSwvQjtH6FQqtJ-55Ksjk_lD9KE1A_lB3L8992rIFm4HwbOQq0N-leMUOEMdnSYJMWqg4ChSH1z9QKM0FJ4LxFlLrghp52mzYMz5TEiqLQEIjDC54K-olDLoSJt12pLF3qgnOfcuVlLcVx37SVakVS2_DJfjkwa56QyNWg7q_KC6nLkajxu5cBS3VF8JUdsPDkgYiZak0DqGF3AQ9YXex8gUeN8V0VoZ94XSmSLROCx2jFYh-kL7n7ghBwKleHX3JCszKlNbVH1DQHEs.OHkvLqQod9OJFifrMgE3qC5zjNu3KJAsL36nx2MZZas&dib_tag=se&keywords=XT90-S%2BConnector%2BAnti-Spark%2BMale%2BFemale%2BConnector%5C&qid=1707853554&s=toys&sprefix=xt90-s%2Bconnector%2Banti-spark%2Bmale%2Bfemale%2Bconnector%2Ctoys%2C219&sr=1-2-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&th=1

**Reply by kelp-videos** [reply]
https://www.amazon.ca/Replacement-Removable-Battery-Compartment-Charging/dp/B0C69S1J4R/ref=sr_1_8?crid=1YTQDHCR38BR0&keywords=gopro+11+outlet+side+door&qid=1707764091&sprefix=gopro+1+outlet+side+door%2Caps%2C162&sr=8-8

**Reply by kelp-videos** [reply]
We just used exacto knives to cut leftover foam to fit the gopros inside of a 4 inch Blue Robotics housing and the bent usb-c connecter to allow the gopro cord to attach within the case.

**Reply by kelp-videos** [reply]
![image_50387969](https://github.com/gopro/labs/assets/157651264/0c9ac394-c008-4b3d-89e1-fd4e69c3e116)
![image_67198465](https://github.com/gopro/labs/assets/157651264/852653bb-35fe-468a-8cde-0878a97895bc)
![image_67210753](https://github.com/gopro/labs/assets/157651264/07522723-7929-4173-a34a-4dff0643d663)
![image_67228929](https://github.com/gopro/labs/assets/157651264/fad91ad1-c905-43ab-94d6-cccf2509974b)

**Reply by caden-fluegge** [reply]
Awesome!  Thanks for such a detailed reply.  Do you have any plans to deal with biofouling of the lens?  We are probably going to end up with a similar rig for our cameras- what end cap did you end up using for the housing?  The GoPro lens sits off-center so we were concerned about optical clarity.  Best of luck for your project!

**Reply by kelp-videos** [reply]
We will just replace the setup after 2 weeks and clean when the divers retrieve it. But open to suggestions, do you have a plan for biofouling? We went with the clear acrylic cap on both sides of the Blue Robotics housing tubes. Best of luck with yours also!

**Reply by caden-fluegge** [reply]
We are currently looking for solutions as our deployments will potentially be months at a time.  So far we have considered a variety of coatings, and may go with ClearSignal.  Seems like there are also a few UV light options as well.
